src/main/java/com/greenfox/tribesoflagopus/backend/service/LocationService.java

```diff
@@ -0,0 +1,37 @@
+package com.greenfox.tribesoflagopus.backend.service;
+
+import com.greenfox.tribesoflagopus.backend.model.entity.Location;
+import com.greenfox.tribesoflagopus.backend.repository.LocationRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class LocationService {
+
+  public static final Integer LOCATION_MIN_VALUE = 1;
+  public static final Integer LOCATION_MAX_VALUE = 100;
+
+  private final LocationRepository locationRepository;
+
+  @Autowired
+  public LocationService(LocationRepository locationRepository) {
+    this.locationRepository = locationRepository;
+  }
+
+  public Location createNewValidLocation() {
+    Location location = new Location();
+    do {
+      location.setX(generateRandomNumber(
+          LOCATION_MIN_VALUE, LOCATION_MAX_VALUE));
+      location.setY(generateRandomNumber(
+          LOCATION_MIN_VALUE, LOCATION_MAX_VALUE));
+    } while (locationRepository.existsByXAndY(location.getX(), location.getY()));
```

Comment:
It's OK in this stage, but keep in mind that if we'll have more and more player joining the game, this method will take more and more time to succeed.
So if we'd like to have a more efficient solution, probably we should get the free locations first, then simply occupy one of those.